alf/networks/transformer_networks.py

```diff
@@ -0,0 +1,175 @@
+# Copyright (c) 2020 Horizon Robotics and ALF Contributors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import gin
+import torch
+import torch.nn as nn
+
+import alf
+from alf.networks import PreprocessorNetwork
+from alf.networks.memory import FIFOMemory
+from alf.nest.utils import NestConcat
+
+
+@gin.configurable
+class TransformerNetwork(PreprocessorNetwork):
+    """A Network composed of Memory and TransformerBlock.
```

Comment:
i think it's easy to confuse the memory of this network with the memory concept of a TransformerBlock. prememory layers still use TransformerBlock's memory, just not the memory created by this network. These two concepts can be better differentiated.

Reply:
> For TransformerBlock, memory is just an argument. TransformerBlock itself does not have any memory.

Yeah that's the reason why it can get confusing because the first argument of line 34 is also called "memory" inside the TransformerBlock.

Reply:
Added comments about this.